src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,72 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+    public static final Map<Class<?>, Class<?>> interfaceImplementation = new HashMap<>();
+    private static final Map<Class<?>, Object> instances = new HashMap<>();
     private static final Injector injector = new Injector();
 
+    static {
+        interfaceImplementation.put(ProductService.class, ProductServiceImpl.class);
+        interfaceImplementation.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementation.put(FileReaderService.class, FileReaderServiceImpl.class);
+    }
+
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Object clazzImplInstance = createNewInstance(clazz);
+        Field[] declaredFields = clazz.getDeclaredFields();
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed, missing @Component annotation on the "
+                    + "class " + clazz.getName());
+        }
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Cant initialize field value. "
+                            + "Class: " + clazz.getName() + ". Field: " + field.getName());
```

Comment:
also a point from common mistakes
```suggestion
                    throw new RuntimeException("Cant initialize field value. "
                            + "Class: " + clazz.getName() + ". Field: " + field.getName(), e);
```

Reply:
done